cw-orch/src/interchain/interchain_channel_builder.rs

```diff
@@ -0,0 +1,271 @@
+
+use crate::ChainState;
+use crate::ContractInstance;
+use crate::Daemon;
+use ibc_chain_registry::chain::ChainData;
+use crate::networks::parse_network;
+use crate::InterchainInfrastructure;
+use crate::follow_ibc_execution::follow_trail;
+use crate::daemon::queriers::DaemonQuerier;
+use ibc_chain_registry::chain::Grpc;
+use ibc_relayer_types::core::ics24_host::identifier::ChainId;
+use tonic::transport::Channel;
+use crate::daemon::channel::GrpcChannel;
+use crate::interchain::IcResult;
+use crate::interchain::docker::DockerHelper;
+use crate::interchain::interchain_channel::InterchainChannel;
+use crate::queriers::Ibc;
+use crate::{DaemonError, Hermes};
+
+use super::interchain_channel::InterchainPort;
+
+
+#[derive(Default)]
+struct ChainChannelBuilder{
+	pub chain_id: Option<String>,
+	pub port: Option<String>,
+	pub grpc_channel: Option<Channel>,
+	pub grpc: Option<Grpc>,
+	pub is_local_chain: Option<bool>
+}
+
+
+impl ChainChannelBuilder{
+	async fn create_grpc_channel(&self) -> Result<Channel, DaemonError>{
+
+		if let Some(channel) = &self.grpc_channel{
+			return Ok(channel.clone());
+		}
+
+		if let Some(grpc) = &self.grpc{
+			return GrpcChannel::connect(&[grpc.clone()], &ChainId::from_string(&self.chain_id.clone().unwrap())).await;
+		}
+
+		// If the GRPC is not registered, we need a default way to query a grpc from a the local config
+	    let mut chains: Vec<ChainData> = vec![parse_network(&self.chain_id.clone().unwrap()).into()];
```

Comment:
This will query the networks details from the chain_id in the registered list cw-orch stores. 
Sometimes, we need to customize this list, especially when working with local docker chains. 
Today, we use a `is_local_chain` flag, but this could be done in a more modular way ?